frontend_tests/zjsunit/zjquery.js

```diff
@@ -284,6 +288,10 @@ exports.make_new_elem = function (selector, opts) {
                             parents_selector + ' in ' + selector);
             return result;
         },
+        prepend: function (arg) {
+            html = arg + html;
+            return self;
+        },
```

Comment:
@showell can you review these test framework changes?

Reply:
I am fine with these additions to `zjquery`, although I kinda wonder if we actually need `append` and `prepend` for the client-side manipulation here?  Instead of appending/prepending around the spoiler header, I wonder if it makes more sense for the servers-side markdown to provide the correct structure in the first place?  And then we can use `.html()` to replace the guts?  This feels like it would be less brittle.

Reply:
I think we want to keep the server-provided HTML relatively simple, since that's very hard to change; one can imagine significant styling changes that we can make with the current model that would be harder if we put the HTML for the button UI in the rendered markdown HTML.